source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -145,14 +145,18 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
             argData.getFlagArgument(kOgsXmlFlag, 0, ogsXmlFileName);
         }
 
+        mx::DocumentPtr document = MaterialXMaya::loadDocument(
```

Comment:
The document is now created outside of the MaterialXData constructor because the latter is often called without recreating the document.

Reply:
Are you talking about changing the element in the document without reloading? That's the only possible "edit" case I can think of. For the longer term it makes sense to cache the document for element value changes. I had a talk with Alvaro+Jonathan and it would be "nice" to just receive more specific value "dirty" updates from them, but not for now at least.

Reply:
Yes, that's the only such use case right now, although I was thinking that if we exposed inputs as attributes, editing those could be similar.